beacon_node/http_api/src/lib.rs

```diff
@@ -1487,29 +1487,44 @@ pub fn serve<T: BeaconChainTypes>(
         .and(warp::path("beacon"))
         .and(warp::path("blob_sidecars"))
         .and(block_id_or_err)
+        .and(warp::query::<api_types::BlobIndicesQuery>())
         .and(warp::path::end())
         .and(chain_filter.clone())
         .and(warp::header::optional::<api_types::Accept>("accept"))
+
         .and_then(
             |block_id: BlockId,
+             indices: api_types::BlobIndicesQuery,
              chain: Arc<BeaconChain<T>>,
              accept_header: Option<api_types::Accept>| {
                 async move {
                     let blob_sidecar_list = block_id.blob_sidecar_list(&chain).await?;
-
+                    let blob_sidecar_list_indexed = match indices.indices {
+                        Some(vec) => {
+                            let mut list = Vec::new();
+                            for i in vec.iter() {
+                                list.push(blob_sidecar_list.get(*i as usize).unwrap().clone());
+                            }
```

Comment:
A few suggestions:
1 . Blobs are quite big so we'd want to avoid cloning as much as possible. Instead of iterating using `iter`, you could use [`.into_iter`](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html) (which moves and consumes the original values), and then use [`.filter`](https://doc.rust-lang.org/std/iter/struct.Filter.html) to retain the requested indices, and finally collect the blob sidecar(s) into a vec using [`.collect`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect).
2. Always avoid using `unwrap` in production code, as it can `panic` and cause Lighthouse to crash. Generally we handle the error by either logging or returning an error.

Reply:
With this current logic, if user sends `1,1,1` as inputs, it would result in the endpoint returning 3 blob sidecars. But I think it you go with using the above approach and iterate using `blob_sidecar_list.into_iter()`, this shouldn't be an issue.

A small bonus would be to move this filtering logic into a function in `block_id.rs`, to keep this already huge file from growing further.

Reply:
Thanks for the feedback !